src/airstack/generic.py

```diff
@@ -225,3 +234,465 @@ def remove_unused_variables(document_ast, query):
+
+        existing_user_index = -1
+        for index, recommended_user in enumerate(recommended_users):
+            recommended_user_addresses = recommended_user.get('addresses', [])
+            if any(addr in recommended_user_addresses for addr in addresses):
+                existing_user_index = index
+                break
+
+        image = content_value.get('image', {}).get(
+            'extraSmall') if content_value.get('image') else None
+
+        new_poap = {
+            'name': name,
+            'image': image,
+            'eventId': event_id
+        }
+
+        if existing_user_index != -1:
+            recommended_user = recommended_users[existing_user_index]
+            _addresses = set(recommended_user.get('addresses', []))
+            _addresses.update(addresses)
+            recommended_user['addresses'] = list(_addresses)
+
+            _poaps = recommended_user.get('poaps', [])
+            if event_id and all(poap['eventId'] != event_id for poap in _poaps):
+                _poaps.append(new_poap)
+            recommended_user['poaps'] = _poaps
+        else:
+            new_user = attendee.get('owner', {})
+            new_user['poaps'] = [new_poap]
+            recommended_users.append(new_user)
+
+    return recommended_users
+
+
+def format_farcaster_followings_data(followings, existing_user=None):
+    if existing_user is None:
+        existing_user = []
+
+    recommended_users = existing_user.copy()
+    for following in followings:
+        existing_user_index = -1
+        for index, recommended_user in enumerate(recommended_users):
+            recommended_user_addresses = recommended_user.get('addresses', [])
+            if any(addr in recommended_user_addresses for addr in following.get('addresses', [])):
+                existing_user_index = index
+                break
+
+        mutual_follower = following.get('mutualFollower', {})
+        follower = mutual_follower.get(
+            'Follower') if mutual_follower is not None else []
+        follows_back = bool(follower[0]) if follower else False
+
+        if existing_user_index != -1:
+            follows = recommended_users[existing_user_index].get('follows', {})
+            recommended_users[existing_user_index] = {
+                **following,
+                **recommended_users[existing_user_index],
+                'follows': {
+                    **follows,
+                    'followingOnFarcaster': True,
+                    'followedOnFarcaster': follows_back
+                }
+            }
+        else:
+            recommended_users.append({
+                **following,
+                'follows': {
+                    'followingOnFarcaster': True,
+                    'followedOnFarcaster': follows_back
+                }
+            })
+
+    return recommended_users
+
+
+def format_lens_followings_data(followings, existing_user=None):
+    if existing_user is None:
```

Comment:
Code duplication
```
    if existing_user is None:
        existing_user = []

    recommended_users = existing_user.copy()
    for following in followings:
        existing_user_index = -1
        for index, recommended_user in enumerate(recommended_users):
            recommended_user_addresses = recommended_user.get('addresses', [])
            if any(address in recommended_user_addresses for address in following.get('addresses', [])):
                existing_user_index = index
                break

        mutual_follower = following.get('mutualFollower', {})
        follower = mutual_follower.get(
            'Follower', []) if mutual_follower is not None else []
        follows_back = bool(follower[0]) if follower else False

        if existing_user_index != -1:
            follows = recommended_users[existing_user_index].get('follows', {})
            recommended_users[existing_user_index].update({
                **following,
                'follows': {
                    **follows,
                    'followingOnLens': True,
                    'followedOnLens': follows_back
                }
            })
        else:
            recommended_users.append({
                **following,
                'follows': {
                    'followingOnLens': True,
                    'followedOnLens': follows_back
                }
            })
         ```
         
         to be taken to another sub method and be used then